### PR TITLE
Split AMI build and validation steps, implement launch permission grant capability

### PR DESCRIPTION
**Pull request by dobbymoodge**

### CODE REVIEW

**Comment by dobbymoodge** [comment]
@stevekuznetsov Does this look sane so far?

**Comment by dobbymoodge** [comment]
Depends on https://github.com/openshift/origin-ci-tool/pull/159

File: sjb/config/test_cases/ami_build_origin_int_rhel_base.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+parent: 'common/test_cases/ami_build.yml'
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/ami_build_origin_int_rhel_base.yml]: you need the rest of config to be in extensions then

Comment (dobbymoodge) [reply sjb/config/test_cases/ami_build_origin_int_rhel_base.yml]: @stevekuznetsov doing that, I get this error:

```
[aos-cd-jobs (grant_launch_opts)]$ cp sjb/config/test_cases/ami_build_origin_int_rhel_base.yml sjb/config/test_cases/jolamb_test_ami_build_origin_int_rhel_base.yml
[aos-cd-jobs (grant_launch_opts)]$ cd -
/$CODE/aos-cd-jobs/sjb
[sjb (grant_launch_opts)]$ python -m generate config/test_cases/jolamb_test_ami_build_origin_int_rhel_base.yml "test" "xml"
[INFO] Generating configuration for test job jolamb_test_ami_build_origin_int_rhel_base
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/$CODE/aos-cd-jobs/sjb/generate.py", line 83, in <module>
    job_config = load_configuration(job_config_path)
  File "/$CODE/aos-cd-jobs/sjb/generate.py", line 55, in load_configuration
    if isinstance(parent_config[entry], list):
KeyError: 'artifacts'
```

Comment (stevekuznetsov) [reply sjb/config/test_cases/ami_build_origin_int_rhel_base.yml]: Your parent needs to have all the fields -- or add defaulting logic to the generator.

Comment (dobbymoodge) [reply sjb/config/test_cases/ami_build_origin_int_rhel_base.yml]: Ok, this is fixed. Let me know if the changes don't quite tickle your pickle.

File: sjb/config/common/test_cases/ami_build.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+provision:
+  os: "fedora"
```
Comment (stevekuznetsov) [inline sjb/config/common/test_cases/ami_build.yml]: RHEL?

**Pull request merged**